Server/index.js

```diff
@@ -7,22 +7,43 @@ const app = express();
 app.use(cors());
 
 app.listen(4000, () => {
-	console.log('Server Works !!! At port 4000');
+    console.log('Server Works !!! At port 4000');
 });
 
-app.get('/downloadmp3', (req,res) => {
-	var url = req.query.url;
-	res.header('Content-Disposition', 'attachment; filename="audio.mp3"');
-	ytdl(url, {
-		format: 'mp3',
-		filter: 'audioonly'
-	}).pipe(res);
+app.get('/downloadmp3', async (req, res, next) => {
+    try {
+        var url = req.query.url;
+        let title = 'audio';
+        await ytdl.getBasicInfo(url, { format: 'mp4' }, (err, info) => {
+            title = info.player_response.videoDetails.title;
+        });
+
+        res.header('Content-Disposition', `attachment; filename='${title}.mp3'`);
```

Comment:
Ok there is a problem here. If you replace the double quotation marks for `attachment; filename='${title}.mp3'` and try to download the file, the server will serve you a `Example.mp3'` rather than `Example.mp3`. Only thing you have to do is replace `attachment; filename='${title}.mp3'` with `attachment; filename="${title}.mp3"` and do the same thing for `attachment; filename='${title}.mp4'